models/migrate.go

```diff
@@ -92,6 +98,97 @@ func insertIssue(ctx context.Context, issue *issues_model.Issue) error {
 	return nil
 }
 
+// UpsertIssues creates new issues and updates existing issues in database
+func UpsertIssues(issues ...*issues_model.Issue) error {
+	ctx, committer, err := db.TxContext()
+	if err != nil {
+		return err
+	}
+	defer committer.Close()
+
+	for _, issue := range issues {
+		if _, err := upsertIssue(ctx, issue); err != nil {
+			return err
+		}
+	}
+	return committer.Commit()
+}
+
+func updateIssue(ctx context.Context, issue *issues_model.Issue) error {
+	sess := db.GetEngine(ctx)
+	if _, err := sess.NoAutoTime().ID(issue.ID).Update(issue); err != nil {
+		return err
+	}
+	issueLabels := resolveIssueLabels(issue.ID, issue.Labels)
+	if len(issueLabels) > 0 {
+		// delete old labels
+		if _, err := sess.Where("issue_id = ?", issue.ID).Delete(); err != nil {
+			return err
+		}
+		// insert new labels
+		if _, err := sess.Insert(issueLabels); err != nil {
+			return err
+		}
```

Comment:
Here we can do:
- delete all issueLabels and then recreate them (current approach)
- use another query beforehand to get existing issueLabels, then delete ones that should be deleted and insert ones that should be inserted

I'm not sure which approach works better. Since number of labels associated with an issue is likely small, I'm guessing delete-all-and-recreate approach will have a smaller overhead than an additional query. But still, I would like to know what everyone thinks.